File: JavaScript/8js 1/cannonsReady.js
/*
|--------------------------------------------------------------------------
| Pirates!! Are the Cannons ready!??
|--------------------------------------------------------------------------
|
*/

function cannonsReady(gunners) {
   return Object.values(gunners).includes("nay") ? "Shiver me timbers!" : "Fire!";
}


const cannonsReady = (gunners) =>
   Object.values(gunners).every((val) => val == "aye") ? "Fire!" : "Shiver me timbers!";


function cannonsReady(gunners) {
   for (var i in gunners) {
      if (gunners[i] == "nay") {
         return "Shiver me timbers!";
      }
   }
   return "Fire!";
}


let a = { Mike: "aye", Joe: "aye", Johnson: "aye", Peter: "aye" };
let b = { Mike: "aye", Joe: "nay", Johnson: "aye", Peter: "aye" };
console.log(cannonsReady(a), "Fire!");
console.log(cannonsReady(b), "Shiver me timbers!");

File: Python/8py 1/cannons_ready.py
"""
|--------------------------------------------------------------------------
| Pirates!! Are the Cannons ready!??
|--------------------------------------------------------------------------
|
"""

def cannons_ready(gunners):
   return "Shiver me timbers!" if "nay" in gunners.values() else "Fire!"


def cannons_ready(gunners):
   for i in gunners:
      if gunners[i] == 'nay':
         return 'Shiver me timbers!'
   return 'Fire!'


def cannons_ready(g):
   return ['Fire!','Shiver me timbers!']['nay' in g.values()]


a = {'Mike':'aye','Joe':'aye','Johnson':'aye','Peter':'aye'}
b = {'Mike':'aye','Joe':'nay','Johnson':'aye','Peter':'aye'}
print(cannons_ready(a),'Fire!')
print(cannons_ready(b),'Shiver me timbers!')